packages/hub/src/utils/symlink.spec.ts

```diff
@@ -1,63 +1,89 @@
-import { describe, test, expect, vi, beforeEach } from "vitest";
-import * as fs from "node:fs/promises";
+/* eslint-disable @typescript-eslint/consistent-type-imports */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { describe, expect, it, vi } from "vitest";
 import { createSymlink } from "./symlink";
-import * as path from "node:path";
-import type { FileHandle } from "node:fs/promises";
-
-vi.mock("node:fs/promises", () => ({
-	rm: vi.fn(),
-	symlink: vi.fn(),
-	rename: vi.fn(),
-	copyFile: vi.fn(),
-	mkdir: vi.fn(),
-	mkdtemp: vi.fn(),
-	open: vi.fn(),
-}));
+import { readFileSync, writeFileSync } from "node:fs";
+import { lstat, rm } from "node:fs/promises";
+import { tmpdir } from "node:os";
+import { join } from "node:path";
+
+let failSymlink = false;
+vi.mock("node:fs/promises", async (importOriginal) => ({
+	...(await importOriginal<typeof import("node:fs/promises")>()),
+	symlink: async (...args: any[]) => {
+		if (failSymlink) {
+			failSymlink = false;
```

Comment:
Not sure to get why we are changing the flag here

Reply:
It's to reset the function once it's called.  (so it doesn't impact other tests)

Could be done more elegantly